### PR TITLE
Add Rust jenkins agent

### PR DESCRIPTION
**Pull request by pabrahamsson**
#### What is this PR About?
A Jenkins agent with Rust toolchain

#### How do we test this?
Provide commands/steps to test this PR.

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

**Comment by pabrahamsson** [comment]
Could maybe @etsauer or @InfoSec812 take a look at this?

File: jenkins-slaves/jenkins-slave-rust/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM quay.io/openshift/origin-jenkins-agent-base:4.2
```
Comment (garethahealy) [inline jenkins-slaves/jenkins-slave-rust/Dockerfile:1]: bump to 4.3?

**Comment by garethahealy** [comment]
@pabrahamsson ; another conflict due to latest fixes. 

code wise, lgtm apart from base image version bump. I've not got any rust exp, so will let other reviews give input.

File: jenkins-slaves/jenkins-slave-rust/Dockerfile (hunk 1 above)
Comment (pabrahamsson) [reply jenkins-slaves/jenkins-slave-rust/Dockerfile]: yep

**Comment by pabrahamsson** [comment]
Thank you @garethahealy, updated the base image and rebased on master, CI is passing.

File: jenkins-slaves/jenkins-slave-rust/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+```bash
+oc process -f ../../.openshift/templates/jenkins-slave-generic-template.yml \
+    -p NAME=jenkins-slave-rust \
+    -p SOURCE_CONTEXT_DIR=jenkins-slaves/jenkins-slave-rust \
+    | oc create -f -
+```
+For all params see the list in the `../../.openshift/templates/jenkins-slave-generic-template.yml` or run `oc process --parameters -f ../../.openshift/templates/jenkins-slave-generic-template.yml`.
+
+## Jenkins
+Add a new Kubernetes Container template called `jenkins-slave-rust` (if you've build and pushed the container image locally) and specify this as the node when running builds. If you're using the template attached; the `role: jenkins-slave` is attached and Jenkins should automatically discover the slave for you. Further instructions can be found [here](https://docs.openshift.com/container-platform/3.7/using_images/other_images/jenkins.html#using-the-jenkins-kubernetes-plug-in-to-run-jobs).
```
Comment (garethahealy) [inline jenkins-slaves/jenkins-slave-rust/README.md:20]: @pabrahamsson ; should the doc reference 3.7? i'd of expected 3.11 as lowest version

Comment (pabrahamsson) [reply jenkins-slaves/jenkins-slave-rust/README.md]: Updated

**Comment by pabrahamsson** [comment]
Just a note to make Tide happy

**Pull request merged**